website/docs/developer/research/research-graphql.md

```diff
@@ -118,18 +118,105 @@ Details on creating and configuring a GraphQL API using AppSync can be found [he
+  ## Additional changes are expected.
+
+# --------------------copy_files_to_archive--------------------------------
+
+  ## shared_recovery.update_status_for_file(), shared_recovery.post_entry_to_queue() functions for sending to SQS might need to be removed and replaced with code for writing to the DB.
+  ## Additional changes are expected.
+
+
+# --------------------request_files----------------------------------------
+
+  ## shared_recovery.update_status_for_file(), shared_recovery.create_status_for_job(), shared_recovery.post_entry_to_queue() function for sending to SQS might need to be removed and replaced with code for writing to the DB.
+  ##  "db_queue_url" arg in inner_task() will not be needed if SQS is not used.
+  ## Modify process_granule() function.
+  ## Additional changes are expected.
+
+# --------------------request_status_for_granule----------------------------------------
+
+  ## Modify get_most_recent_job_id_for_granule() function to use GraphQL query.
+  ## Modify get_status_totals_for_job() function to use GraphQL query.
+  ## Additional changes are expected.
+
+# --------------------request_status_for_job----------------------------------------
+
+  ## Modify get_granule_status_entries_for_job() function to use GraphQL query.
+  ## Modify get_granule_status_entries_for_job() function to use GraphQL query.
+
+# --------------------db_deploy----------------------------------------
+
+  ## app_db_exists(), app_schema_exists(), app_version_table_exists(), get_migration_version() functions might need to be updated.
+  ## Additional changes are expected.
+
+# --------------------post_to_database----------------------------------------
+
+  ## This will need to be removed if GraphQL is used since it will write to the DB instead of SQS.
+
+```
+
+### GraphQL server recommendation
+
+Based on this research, GraphQL has a higher learning curve compared to other technologies and will take some time for developers to learn and then implement. If using Javascript libraries, developers should have good background in this language in order to execute this approach. Using some prebuilt GraphQL servers that automatically generates GraphQL schema and resolvers could save some time and simplify the design. Building a prototype in ORCA could reveal if it is worth the effort and time. However, using lambda, API gateway and SQS(the resources in existing ORCA architecture) seem to contain more resources and examples online than GraphQL.
+
+#### Recommendation #1- [Hasura](https://hasura.io/)
+
+- Hasura GraphQL engine can be deployed using using Docker and Docker Compose or using Hasura cloud. 
+- The easiest way to set up Hasura GraphQL engine on local environment without any cost is using Docker. 
+- It supports GraphQL on Postgres, AWS Aurora and seems to be compatible with the current ORCA architecture. 
+- Cost to use the cloud is $99/month/project and supports upto 20GB data with $2/additional GB data. 
```

Comment:
Is this cost based on using Hasura Cloud or a supported version of the software? Is the paid version still part of AWS? If the paid version for support or SaaS? If it is SaaS it is approved on NGAP?

Reply:
cost is for using Hasura cloud. 
`https://hasura.io/pricing/`

Looks like you have to pay on their website, not on AWS
Paid version is for SaaS and I am not sure if it is approved by NGAP. I have not heard of anyone using it so far in NGAP.